### PR TITLE
[MM-41263] Fix stackoverflow error in saml login

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR fixes the stackoverflow error in SAML login by reducing the number of bytes passed to SAML service provider.
Notice that 2MB is an arbitrary number for the SAMLResponse size.

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-41263

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: web/saml.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/mattermost/mattermost-server/v6/utils"
 )
 
+const maxSAMLResponseSize = 2 * 1024 * 1024 // 2MB
```
Comment (jupenur) [inline web/saml.go:19]: I'm not super confident on what this number should actually be, but the few points of comparison that I can find -- SAML SP implementations that set a limit on Response or Assertion lengths -- all set the limit to a few hundred kilobytes. Based on that, 2 MB should be plenty enough for interoperability.

**Review by jupenur — APPROVED** [review]
LGTM

**Comment by amyblais** [comment]
@Willyfrog @anurag6713 This is a backport candidate.

**Comment by amyblais** [comment]
@iomodo Does this need QA testing?

**Comment by iomodo** [comment]
@amyblais I did test, but second pair of eyes might be good.

**Review by cpoile — APPROVED** [review]
nice fix!

**Comment by amyblais** [comment]
@furqanmlk This is a backport candidate.

**Comment by furqanmlk** [comment]
> https://mattermost.atlassian.net/browse/MM-41263

@iomodo Can you please provide QA steps ?
I just don’t want to miss the actual issue

**Comment by iomodo** [comment]
@furqanmlk there are too things to check
1. Big data should not cause server to crash.
2. You should be able to login with SAML.

**Review by furqanmlk — APPROVED** [review]
### Logged in Successfully 

image

### Big Data Script Passed

image

**Pull request merged**